src/talk.ts

```diff
@@ -16,16 +19,22 @@ export const talk = async (prompt: string, input: string, llamaServerUrl: string
     if (sentenceEndRegex.test(token)) {
       const sentence = currentSentence.join('');
       sentences.push(sentence);
-      const promise = generateAudio(sentence);
       promisesChain = promisesChain.then(async () => { 
-        await promise.then(playAudioFile);
-        sentenceCallback(sentence);
+        const release = await mutex.acquire();
+        try {
+          const audioFilePath = await generateAudio(sentence);
+          await playAudioFile(audioFilePath);
+          sentenceCallback(sentence);
+        } catch (err) {
+          console.log('Error generating audio');
+        } finally {
+          release();
+        }
```

Comment:
Ensuring that two responses aren't happening at the same time likley should be handled upstream. Maybe it's something that can be put into the responsereflex, using a new event?

Reply:
Though about this but not sure other than to add mutex upstream.